packages/flutter_tools/lib/src/web/compiler_config.dart

```diff
@@ -134,9 +130,9 @@ class JsCompilerConfig extends WebCompilerConfig {
 /// Configuration for the Wasm compiler.
 class WasmCompilerConfig extends WebCompilerConfig {
   const WasmCompilerConfig({
-    required this.omitTypeChecks,
-    required this.wasmOpt,
-    required super.renderer
+    this.omitTypeChecks = false,
```

Comment:
I thought `omitTypeChecks` was like the "turbo" button... do we want this always off by default? :P

Reply:
Yeah... I'm still not 100% sure it's the right default but at this point it seems like most use cases turn it on, so I'll make it `true` by default.

Reply:
Actually, to be honest, I think I'd still rather leave the default as `false` so that it matches the default of the compiler itself for now. The way this is specified might actually be changing soon anyway since we'll be using `dart compile wasm` and we will have `-On` optimization levels that encapsulate this.